### DISCUSSION TITLE
JSNAPY timeout error

### DISCUSSION THREAD

**Discussion by rjangawl** [community/community#45776] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi Team, 

Need help with below mentioned issue wherein ‘show interface extensive’ command has issues when taking a snapshot in Jsnapy.
 
root@control-node:/home/telecomegypt/jsnapy/testfiles/Jsnapy-testfiles# jsnapy --snap pre -f  config.yml --folder /home/telecomegypt/jsnapy/
 
--- <snipped> ---
 
Connecting to device 1.1.1.1 ................
Taking snapshot of COMMAND: show chassis hardware
Taking snapshot of COMMAND: show chassis fpc pic-status
Taking snapshot of COMMAND: show interfaces extensive
Exception in thread Thread-49:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/jsnapy.py", line 697, in connect
    dev, output_file, hostname, config_data, **kwargs
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/jsnapy.py", line 419, in generate_rpc_reply
    val = g.generate_reply(tests, dev, output_file, hostname, self.db)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/snap.py", line 434, in
generate_reply
    self.run_cmd(test_file, t, formats, dev, output_file, hostname, db)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/snap.py", line 182, in run_cmd
    etree.tostring(err.rsp, encoding="unicode"),
  File "src/lxml/etree.pyx", line 3462, in lxml.etree.tostring
TypeError: Type 'NoneType' cannot be serialized.
 
Connecting to device 2.2.2.2 ................
Taking snapshot of COMMAND: show chassis hardware
Taking snapshot of COMMAND: show chassis fpc pic-status
Taking snapshot of COMMAND: show chassis environment
Taking snapshot of COMMAND: show chassis routing-engine
Taking snapshot of COMMAND: show version invoke-on all-routing-engines
Taking snapshot of COMMAND: show chassis alarms
Taking snapshot of COMMAND: show system core-dumps routing-engine both
Taking snapshot of COMMAND: show arp no-resolve
Taking snapshot of COMMAND: show arp no-resolve
Taking snapshot of COMMAND: show interfaces terse
Taking snapshot of COMMAND: show route summary
Taking snapshot of COMMAND: show route forwarding-table summary
Taking snapshot of COMMAND: show bfd session extensive
Taking snapshot of COMMAND: show ospf neighbor
Taking snapshot of COMMAND: show ospf interface
Taking snapshot of COMMAND: show bgp summary
Taking snapshot of COMMAND: show ospf database detail
Exception in thread Thread-53:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/jsnapy.py", line 697, in connect
    dev, output_file, hostname, config_data, **kwargs
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/jsnapy.py", line 419, in generate_rpc_reply
    val = g.generate_reply(tests, dev, output_file, hostname, self.db)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/snap.py", line 434, in generate_reply
    self.run_cmd(test_file, t, formats, dev, output_file, hostname, db)
  File "/usr/local/lib/python2.7/dist-packages/jnpr/jsnapy/snap.py", line 182, in run_cmd
    etree.tostring(err.rsp, encoding="unicode"),
  File "src/lxml/etree.pyx", line 3462, in lxml.etree.tostring
TypeError: Type 'NoneType' cannot be serialized.
 
Connecting to device 3.3.3.3 ................
 
--- <snipped> ---
 
If I remove the .yml file related to this command (show interfaces extensive), then it works fine. These issues were seen only for MX104 router as it has a larger scale and router takes time to respond to these commands even via CLI. Commands like ‘show ospf database detail’ also get stuck when running via Jsnapy.
 
telecomegypt@MX104-RE0:MX104-PE2> show ospf database detail | match Topology | count
Count: 15363 lines
 
telecomegypt@MX104-RE0:MX104-PE2> show interfaces terse | match "up    up" | count
Count: 3 lines
 
telecomegypt@MX104-RE0:MX104-PE2> clear cli logical-system
Cleared default logical system
 
telecomegypt@MX104-RE0> show interfaces terse | match "up    up" | count
Count: 4472 lines
telecomegypt@MX104-RE0>
 
 
We need help to increase the timeout during pre/post snapshot. Attached are the config and device list files used in our case.

**Comment by queenofcorgis** [comment, marked as answer]
If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Reply by rjangawl** [reply]
Thanks the page does have a contact details for this tool - [jnpr-community-netdev@juniper.net](mailto:jnpr-community-netdev@juniper.net)

will reach out to them.

**Comment by cristiano54321** [comment]
Hi @rjangawl.

Did you find any solution to this problem?

Thanks.
